### PR TITLE
updated documentation regarding behavior of `rbindlist` when applied to `difftime` objects with different units

### LINKED ISSUE #4541: rbindlist gives wrong results when it is applied to difftime objects with different units.
`#` [`Minimal reproducible example`](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example)

```
library(data.table)

difftime_diffunits <- list(
  structure(list(
    start_time = "2020-06-02 05:12:14.652", end_time = "2020-06-02 06:01:24.753", 
    diff_time = structure(49.1683500011762, class = "difftime", units = "mins")), 
    row.names = c(NA, -1L), class = c("data.table", "data.frame")), 
  structure(list(
    start_time = "2020-06-02 05:12:14.656", end_time = "2020-06-02 06:02:25.045", 
    diff_time = structure(50.1731500029564, class = "difftime", units = "mins")), 
    row.names = c(NA, -1L), class = c("data.table", "data.frame")), 
  structure(list(
    start_time = "2020-06-02 05:12:14.973", end_time = "2020-06-02 06:32:28.081", 
    diff_time = structure(1.33697444445557, class = "difftime", units = "hours")), 
    row.names = c(NA, -1L), class = c("data.table", "data.frame")), 
  structure(list(
    start_time = "2020-06-02 05:12:14.973", end_time = "2020-06-02 06:24:51.367", 
    diff_time = structure(1.21010944445928, class = "difftime", units = "hours")), 
    row.names = c(NA, -1L), class = c("data.table", "data.frame")))

# [[1]]
# start_time                end_time     diff_time
# 1: 2020-06-02 05:12:14.652 2020-06-02 06:01:24.753 49.16835 mins
# 
# [[2]]
# start_time                end_time     diff_time
# 1: 2020-06-02 05:12:14.656 2020-06-02 06:02:25.045 50.17315 mins
# 
# [[3]]
# start_time                end_time      diff_time
# 1: 2020-06-02 05:12:14.973 2020-06-02 06:32:28.081 1.336974 hours
# 
# [[4]]
# start_time                end_time      diff_time
# 1: 2020-06-02 05:12:14.973 2020-06-02 06:24:51.367 1.210109 hours

rbindlist(difftime_diffunits)

# start_time                end_time      diff_time
# 1: 2020-06-02 05:12:14.652 2020-06-02 06:01:24.753 49.168350 mins
# 2: 2020-06-02 05:12:14.656 2020-06-02 06:02:25.045 50.173150 mins
# 3: 2020-06-02 05:12:14.973 2020-06-02 06:32:28.081  1.336974 mins
# 4: 2020-06-02 05:12:14.973 2020-06-02 06:24:51.367  1.210109 mins
```
**Note** that the units of diff_time of the 3rd and 4th rows should be "**hours**" instead of "**mins**".

`#` `Output of sessionInfo()`
```
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows Server x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.12.9

loaded via a namespace (and not attached):
[1] compiler_4.0.0 tools_4.0.0    tinytex_0.23   xfun_0.14     
```

### PR DESCRIPTION
**Pull request by Nj221102**
closes #4541 

#### Summary

The current implementation of `rbindlist` does not preserve the units of `difftime` objects correctly when combining `data.table` or `data.frame` objects with different units. Instead, it converts all `difftime` values to a common unit without adjusting the values themselves. This can lead to incorrect time differences in the resulting `data.table`. 

This pull request proposes updating the documentation to inform users that they need to manually ensure consistent units for `difftime` objects before using `rbindlist` to avoid inaccuracies.

### CODE REVIEW

**Comment by ben-schwen** [comment]
As pointed out by Jan in the issue I don't think that a tailored solution to a certain class is a favorable solution. Adding documentation about a single class is therefore not ideal either.

Can we generalize this to classes where the unit depends on class attributes?

**Comment by MichaelChirico** [comment]
I agree about the need to be more generic here.

**Comment by Nj221102** [comment]
@ben-schwen is this good enough ??

```
When binding lists of \code{data.table} or \code{data.frame} objects containing objects with units defined by class attributes (e.g., \code{difftime} objects with different units), the resulting \code{data.table} may not preserve the original units correctly. Instead, values will be converted to a common unit (typically the smallest unit present in the input) without proper conversion of the values themselves. This issue applies to any class where the unit or precision is determined by attributes. Users should manually ensure that objects with unit-dependent attributes have consistent units before using \code{rbindlist}
```

File: man/rbindlist.Rd [hunk 1]
```diff
@@ -24,6 +24,8 @@ Columns with duplicate names are bound in the order of occurrence, similar to ba
...
+
+When binding lists of \code{data.table} or \code{data.frame} objects containing objects with units defined by class attributes (e.g., \code{difftime} objects with different units), the resulting \code{data.table} may not preserve the original units correctly. Instead, values will be converted to a common unit (typically the smallest unit present in the input) without proper conversion of the values themselves. This issue applies to any class where the unit or precision is determined by attributes. Users should manually ensure that objects with unit-dependent attributes have consistent units before using \code{rbindlist}
```
Comment (ben-schwen) [inline man/rbindlist.Rd:28]: > (typically the smallest unit present in the input)
This is not true, compare e.g.

```r
a = data.table(x = as.difftime("10:00:00"))
b = data.table(x = as.difftime("00:05:00"))
rbind(a,b)
rbind(b,a)
```

Comment (Nj221102) [reply man/rbindlist.Rd]: Removed 👍

File: man/rbindlist.Rd [hunk 2]
```diff
@@ -24,6 +24,8 @@ Columns with duplicate names are bound in the order of occurrence, similar to ba
...
 
 If a column contains factors then a factor is created. If any of the factors are also ordered factors then the longest set of ordered levels are found (the first if this is tied). Then the ordered levels from each list item are checked to be an ordered subset of these longest levels. If any ambiguities are found (e.g. \code{blue<green} vs \code{green<blue}), or any ordered levels are missing from the longest, then a regular factor is created with warning. Any strings in regular factor and character columns which are missing from the longest ordered levels are added at the end.
+
+When binding lists of \code{data.table} or \code{data.frame} objects containing objects with units defined by class attributes (e.g., \code{difftime} objects with different units), the resulting \code{data.table} may not preserve the original units correctly. Instead, values will be converted to a common unit without proper conversion of the values themselves. This issue applies to any class where the unit or precision is determined by attributes. Users should manually ensure that objects with unit-dependent attributes have consistent units before using \code{rbindlist}
```
Comment (ben-schwen) [inline man/rbindlist.Rd:28]:
```suggestion
When binding lists of \code{data.table} or \code{data.frame} objects containing objects with units defined by class attributes (e.g., \code{difftime} objects with different units), the resulting \code{data.table} may not preserve the original units correctly. Instead, values will be converted to a common unit without proper conversion of the values themselves. This issue applies to any class where the unit or precision is determined by attributes. Users should manually ensure that objects with unit-dependent attributes have consistent units before using \code{rbindlist}.
```

**Comment by ben-schwen** [comment]
Ty Nitish!

**Pull request merged**